### ISSUE TITLE
Emojis don't appear on MacOS 11 Big Sur 😥

### ISSUE THREAD

**Issue by ronblum** [rstudio/rstudio#7897]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.1093 and 1.4.869
    OS Version      : MacOS 11 Big Sur beta 8
    R Version       : 4.0.2 

### Steps to reproduce the problem

Method 1
1) Open a notebook or R markdown file
2) Copy this emoji: 😀
3) Paste the emoji in the notebook or R markdown file
   - The only thing that appears is a blank space
4) Copy the blank space
5) Paste it elsewhere, e.g. in Notes
   - The emoji appears as expected: 😀

Method 2
1) Open a notebook or R markdown file
2) Switch to visual editor mode
3) `Insert -> Insert Emoji`
   - Note all the spaces for emojis, and the changing descriptions when hovering over them, but no actual emojis
![image](https://user-images.githubusercontent.com/31009952/94480339-2b70a900-01a4-11eb-90c0-80f88021ae16.png)
4) Click on one of the empty spaces
   - A blank space is inserted in the file
5) Copy the blank space
6) Paste elsewhere, e.g. in Notes
   - An emoji appears, e.g. 😀

### Describe the problem in detail

Emojis are all showing a blank.

### Describe the behavior you expected

Emojis should be displayed as emojis.

As far as I can tell, this is only a problem in Big Sur. I tried it with RStudio Desktop on MacOS 10.15.6 Catalina, and RStudio Server on Red Hat 8.2, and they work fine. 

![image](https://user-images.githubusercontent.com/31009952/94480691-ac2fa500-01a4-11eb-90bc-080648966eee.png)

Note: this isn't a regression—it fails in 1.3.1093, as well as 1.4.869.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by tyluRp** [comment]
This is happening for me as well after updating to `20A5374i`

**Comment by dragonstyle** [comment]
I've taken a quick first look and it appears to be limited to the standalone version of RStudio (running QT). 

* Appears emoji don't render properly anywhere in standalone version on Big Sur (e.g. not in visual or source markdown editor, not in find to other text boxes). Confirmed that they render properly when viewing RStudio in Chrome.
* Confirmed that a saved version of the document that doesn't properly render the emoji when rendered on a non Big-Sur version of MacOS does render the emoji properly (confirming that the character is being written to the document properly).

Still investigating...

**Comment by gtritchie** [comment]
A strategy we often employ in seeing if an issue is in QtWebEngine is to build and run the Qt sample "webengine widgets simple browser" which uses QtWebEngine to build a simple web browser.

If it repros you can then try using a new Qt and see if it happens to be fixed... though us upgrading Qt for 1.4 would be a non-starter given the fact we always, and I mean **always**, find tricky regressions in every Qt update we have taken, major or minor.

**Comment by dragonstyle** [comment]
I just went through this process and tested a simple web page with a ContentEditable div. It appears that the issue exists for QT 5.12.8, but is resolved in later versions of QT (confirmed fixed on 5.13.2 but still narrowing to when precisely it was fixed).

**Comment by gtritchie** [comment]
It would be worth trying 5.12.9 (we are currently on 5.12.8); if that magically fixes it (longshot) it would at least be a possibility to consider it for 1.4 (maybe in a follow up patch release). Going to 5.13, 5.14, or 5.15 would be much riskier.

There's also a 5.12.10 release planned, although Qt's schedule says it was slated for Sept 20, which clearly didn't quite happen:

https://wiki.qt.io/Qt_5.12_Release

**Comment by jmcphers** [comment]
@dragonstyle Were you able to test with Qt 5.12.9? We could experiment with an update to that release on a branch if we know it addresses this issue.

**Comment by dragonstyle** [comment]
I'll give that a quick look,

**Comment by ronblum** [comment]
Addendum: Updating to Big Sur 11 beta 9, 20A5384c, didn't help.

**Comment by dragonstyle** [comment]
It doesn't appear to be resolved in 5.12.9, unfortunately. I'm still seeing the issue in their nanobrowser example. Is it worth me testing more versions of 5.13.x (or other versions)?

FWIW, I'm just viewing this in their nano browser to confirm rendering issue:

```<html>

<body>
<div contenteditable="true" height="500px" width="100%" style="border: solid red 1px;">
emoji->😀<- emoji
</div>

</body>
</html>```

In 5.12.8/9 there will be only whitespace rendered between the arrows...

**Comment by gtritchie** [comment]
Got it (wasn't very optimistic about 5.12.9 but was worth a try). I'm investigating an update to 5.15.1.

**Comment by gtritchie** [comment]
Opened an issue to Qt, in case that reveals something interesting (would be awesome if it was fixed in upcoming 5.12.10).

https://bugreports.qt.io/browse/QTBUG-87689

**Comment by gtritchie** [comment]
After all of that, this appears to have been fixed in latest Big Sur update: 11.0.1 Beta (20B5012d)

Worked for me in both 1.4 daily build, and the 1.3 and 1.2 releases.

**Comment by ronblum** [comment]
![image](https://user-images.githubusercontent.com/31009952/97590079-a2cc6f00-19d4-11eb-83c0-7f9bd39f067d.png)

Verified in RStudio Desktop 1.3.1093 and 1.4.969 on MacOS 11.0.1 beta.
